### ISSUE TITLE
Casing issue with initializeServerAttributes method when reading the edition value

### ISSUE THREAD

**Issue by florisg-infosupport** [liquibase/liquibase-neo4j#556]
I am using the version: **org.liquibase.ext:liquibase-neo4j:4.26**.
The issue occurs when finding/initializig the Neo4jDatabase, based on the connection given from the Neo4jDriver.

The issue: the resuling `Map<string, ?>` from `Map<String, ?> components = this.readComponents();` in the Neo4jDatabase(line 243) has its keys in ALL_CAPS, where it expects them to be in lower case. This results in a null pointer when trying to use the values from the map.

The error:
```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.toLowerCase(java.util.Locale)" because the return value of "java.util.Map.get(Object)" is null
	at liquibase.ext.neo4j.database.Neo4jDatabase.initializeServerAttributes(Neo4jDatabase.java:245)
	at liquibase.ext.neo4j.database.Neo4jDatabase.setConnection(Neo4jDatabase.java:45)
	at liquibase.database.DatabaseFactory.findCorrectDatabaseImplementation(DatabaseFactory.java:121)
```

The components object:
![image](https://github.com/liquibase/liquibase-neo4j/assets/43369750/cc683580-7e4c-47d7-a5b8-25043ecc1f1b)

**Comment by fbiville** [comment]
Hello and thanks for raising the issue.
Can you please confirm the version of Liquibase core you are using alongside liquibase-neo4j?

**Comment by florisg-infosupport** [comment]
No problem,
That is: `org.liquibase:liquibase-core:jar:4.16.1`

**Comment by fbiville** [comment]
Thanks, as I suspected, the version of core is too old for the Neo4j extension's version.

The short answer is: you should upgrade the core version to at least 4.26.0, or downgrade the extension to 4.16.1.

The long explanation is as follows:

Since the release of liquibase-neo4j 4.21.0, the components have indeed been lowercased, and that only works with core >= 4.17.0 ([see second entry of the 4.21.0 release](https://github.com/liquibase/liquibase-neo4j/releases/tag/liquibase-neo4j-4.21.0)).

Moreover:

- liquibase-neo4j 4.23.1 (and later) [requires at least Liquibase core 4.23.1](https://github.com/liquibase/liquibase-neo4j/releases/tag/v4.23.1)
- liquibase-neo4j 4.24.0 (and later) [requires at least Liquibase core 4.24.0](https://github.com/liquibase/liquibase-neo4j/releases/tag/v4.24.0)

The versioning policy of the Neo4j extension is detailed [here](https://www.neo4j.com/labs/liquibase/docs/#liquibase).
In short, the extension version is guaranteed to work with the corresponding core version, but not with earlier or later versions. If there is a strong alignment requirement, the release notes will detail it.

Does that make sense?

**Comment by florisg-infosupport** [comment]
Thanks for the quick replies, that indeed makes sense. In that case we will upgrade the core to resolve the issue.

**Comment by fbiville** [comment]
Perfect, let me know if you run into more issues.
